File: leetcode/118-pascals-triangle/solution.py
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        tri = []
        for i in range(numRows):
            row = [1] * (i + 1)
            for j in range(1, i):
                row[j] = tri[i - 1][j - 1] + tri[i - 1][j]
            tri.append(row)
        return tri


if __name__ == "__main__":
    for row in Solution().generate(5):
        print(row)
    # [1]
    # [1, 1]
    # [1, 2, 1]
    # [1, 3, 3, 1]
    # [1, 4, 6, 4, 1]

File: leetcode/118-pascals-triangle/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> tri(numRows);
        for (int i = 0; i < numRows; i++) {
            tri[i].resize(i + 1, 1);
            for (int j = 1; j < i; j++) {
                tri[i][j] = tri[i - 1][j - 1] + tri[i - 1][j];
            }
        }
        return tri;
    }
};

int main() {
    Solution sol;
    auto tri = sol.generate(5);
    for (auto& row : tri) {
        for (int v : row) cout << v << " ";
        cout << endl;
    }
    return 0;
}
